blockchain-agent/src/main/java/com/sphereon/alfresco/blockchain/agent/tasks/ondemand/VerifyRegistrations.java

```diff
@@ -0,0 +1,84 @@
+import static com.sphereon.alfresco.blockchain.agent.model.AlfrescoBlockchainRegistrationState.REGISTERED;
+
+@Component
+@Scope(ConfigurableBeanFactory.SCOPE_PROTOTYPE)
+public class VerifyRegistrations {
+    private static final Logger logger = LoggerFactory.getLogger(VerifyRegistrations.class);
+
+    private final ObjectFactory<AlfrescoRepository> alfrescoRepositoryFactory;
+    private final VerifyRegistrationTask verificationTask;
+
+    public VerifyRegistrations(final ObjectFactory<AlfrescoRepository> alfrescoRepositoryFactory,
+                               final VerifyRegistrationTask verificationTask) {
+        this.alfrescoRepositoryFactory = alfrescoRepositoryFactory;
+        this.verificationTask = verificationTask;
+    }
+
+    public List<VerifyContentAlfrescoResponse> execute(final List<String> selectedNodeIds,
+                                                       final String credentials) {
+        final List<VerifyContentAlfrescoResponse> contentResponses = new ArrayList<>();
+        try {
+            final var alfrescoRepository = this.alfrescoRepositoryFactory.getObject();
+            if (StringUtils.isNotEmpty(credentials)) {
+                this.updateCredentials(alfrescoRepository, credentials);
+            }
+            alfrescoRepository.selectEntries(selectedNodeIds).stream()
+                    .map(NodeEntry::getEntry)
+                    .filter(entry -> {
+                        final String entryCurrentState = getProperty(entry, "bc:RegistrationState");
+                        return !PENDING_REGISTRATION.getKey().equals(entryCurrentState);
+                    })
+                    .forEach(entry -> {
+                        logger.info("Found document " + entry.getName() + " / " + entry.getId());
+                        final var contentHash = alfrescoRepository.hashEntry(entry.getId());
+                        final var response = verificationTask.verify(contentHash);
+                        contentResponses.add(response);
+                        final var registrationState = response.getRegistrationState();
+                        final var registrationTime = response.getRegistrationTime();
+                        final var singleProofChainChainId = response.getSingleProofChainId();
+                        final var perHashProofChainChainId = response.getPerHashProofChainId();
+                        alfrescoRepository.updateAlfrescoNodeWith(entry.getId(), registrationState, registrationTime, singleProofChainChainId, perHashProofChainChainId);
+                        if (response.getRegistrationState() != REGISTERED) {
+                            logger.info("Document " + entry.getName() + " / " + entry.getId() + " was not registered yet: " + response.getRegistrationState());
+                        }
+                    });
+        } catch (Exception exception) {
+            logger.error("An error occurred whilst executing VerifyRegistrationsTask", exception);
```

Comment:
.getMessage?

Reply:
Well, this will match to `Logger::error(String var1, Throwable var2)`, so we get the stack trace too. I'm not a big fan of blindly catching anything though, so maybe the whole statement can go.

Reply:
Yeah, but if in production you disable stack traces, the .getMessage would tell you something ;).

The broad catch can be removed in the future, but not prio

Reply:
Good point. Fixed.